include/exadg/fluid_structure_interaction_precice/driver_solid.h

```diff
@@ -0,0 +1,333 @@
+      timer_local.restart();
+
+      // setup spatial operator
+      structure_operator = std::make_shared<Structure::Operator<dim, Number>>(
+        this->application->get_grid_structure(),
+        this->application->get_boundary_descriptor_structure(),
+        this->application->get_field_functions_structure(),
+        this->application->get_material_descriptor_structure(),
+        this->application->get_parameters_structure(),
+        "elasticity",
+        this->mpi_comm);
+
+      // initialize matrix_free
+      structure_matrix_free_data = std::make_shared<MatrixFreeData<dim, Number>>();
+      structure_matrix_free_data->append(structure_operator);
+
+      structure_matrix_free = std::make_shared<dealii::MatrixFree<dim, Number>>();
+      structure_matrix_free->reinit(*this->application->get_grid_structure()->mapping,
+                                    structure_matrix_free_data->get_dof_handler_vector(),
+                                    structure_matrix_free_data->get_constraint_vector(),
+                                    structure_matrix_free_data->get_quadrature_vector(),
+                                    structure_matrix_free_data->data);
+
+      structure_operator->setup(structure_matrix_free, structure_matrix_free_data);
+
+      // initialize postprocessor
+      structure_postprocessor = this->application->create_postprocessor_structure();
+      structure_postprocessor->setup(structure_operator->get_dof_handler(),
+                                     *this->application->get_grid_structure()->mapping);
+
+      this->timer_tree.insert({"FSI", "Setup", "Structure"}, timer_local.wall_time());
+    }
+    /**************************************** STRUCTURE *****************************************/
+
+    /**************************************** STRUCTURE *****************************************/
+
+
+    /*********************************** INTERFACE COUPLING *************************************/
+
+    this->precice =
+      std::make_shared<Adapter::Adapter<dim, dim, VectorType>>(this->precice_parameters);
+
+    write_mesh_name = this->precice_parameters.write_mesh_name;
+    this->precice->add_write_interface(
+      this->application->get_boundary_descriptor_structure()->neumann_mortar_bc.begin()->first,
+      this->precice_parameters.write_mesh_name,
+      {this->precice_parameters.write_data_name, "Velocity"},
+      "values_on_dofs",
+      structure_matrix_free,
+      structure_operator->get_dof_index(),
+      numbers::invalid_unsigned_int);
+
+    // structure to ALE
+    // structure to fluid
+
+    // fluid to structure
+    {
+      std::vector<unsigned int> quad_indices;
+      quad_indices.emplace_back(structure_operator->get_quad_index());
+
+      // VectorType stress_fluid;
+      communicator = std::make_shared<InterfaceCoupling<dim, dim, Number>>(this->precice);
+      VectorType displacement_structure;
+      structure_operator->initialize_dof_vector(displacement_structure);
```

Comment:
this variable is not used for the next couple of lines, so the code should be shifted to where it is used.